tests/integration/helm_fluentbit_containerd_multiline_logs_test.go

```diff
@@ -0,0 +1,318 @@
+	log "k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/ctxopts"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/receivermock"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/stepfuncs"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/strings"
+)
+
+func Test_Helm_FluentBit_Containerd_Multiline_Logs(t *testing.T) {
+	const (
+		tickDuration           = 3 * time.Second
+		waitDuration           = 3 * time.Minute
+		logRecords             = 4
+		logLoops               = 500
+		multilineLogCount uint = logRecords * logLoops
+	)
+	var (
+		expectedMetrics = internal.DefaultExpectedMetrics
+	)
+
+	// TODO:
+	// Refactor this: we should find a way to inject this into step func helpers
+	// like stepfuncs.WaitUntilPodsAvailable() instead of relying on an implementation
+	// detail.
+	releaseName := strings.ReleaseNameFromT(t)
+
+	featInstall := features.New("installation").
+		Assess("sumologic secret is created",
+			func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+				k8s.WaitUntilSecretAvailable(t, ctxopts.KubectlOptions(ctx), "sumologic", 60, tickDuration)
+				secret := k8s.GetSecret(t, ctxopts.KubectlOptions(ctx), "sumologic")
+				require.Len(t, secret.Data, 10)
+				return ctx
+			}).
+		Assess("fluentd logs pods are available",
+			stepfuncs.WaitUntilPodsAvailable(
+				v1.ListOptions{
+					LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-logs", releaseName),
+				},
+				3,
+				waitDuration,
+				tickDuration,
+			),
+		).
+		Assess("fluentd logs buffers PVCs are created",
+			func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+				namespace := ctxopts.Namespace(ctx)
+				releaseName := ctxopts.HelmRelease(ctx)
+				kubectlOptions := ctxopts.KubectlOptions(ctx)
+
+				t.Logf("kubeconfig: %s", kubectlOptions.ConfigPath)
+				cl, err := k8s.GetKubernetesClientFromOptionsE(t, kubectlOptions)
+				require.NoError(t, err)
+
+				assert.Eventually(t, func() bool {
+					pvcs, err := cl.CoreV1().PersistentVolumeClaims(namespace).
+						List(ctx, v1.ListOptions{
+							LabelSelector: fmt.Sprintf("app=%s-sumologic-fluentd-logs", releaseName),
+						})
+					if !assert.NoError(t, err) {
+						return false
+					}
+
+					return err == nil && len(pvcs.Items) == 3
+				}, waitDuration, tickDuration)
+				return ctx
+			}).
+		Assess("fluentd metrics pods are available",
```

Comment:
I'd remove all the code related to metrics both from the test code and from values.yaml.

This is already tested elsewhere. Just leave here whatever is related to containerd multiline logs.

I'd also remove the tests related to logs as that is tested also in a separate test.

Reply:
done